cds/modules/search/access_control.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+""" Search access control. """
+
+from elasticsearch_dsl.query import Q
+from flask import session, current_app
+from flask_login import current_user
+from functools import partial
+
+from invenio_records_rest.query import default_search_factory
+
+###############################################################################
+#  Records should have an '_access' field containing three nested fields,
+#  namely 'read', 'write' and 'admin', each having zero or more groups/emails
+#  that are given the enclosing access right.
+#
+#  JSON example:
+#
+#   "_access": [
+#      {
+#       "read": ["it-dep-cda", "it-dep", "reader1@cern.ch", "reader2@cern.ch"],
+#       "write": ["it-dep-cda", "writer@cern.ch"],
+#       "admin": ["orestis.melkon@cern.ch"]
+#      }
+#   ],
+#
+###############################################################################
+
+
+def cern_query_parser(query):
+    """Restrict search results based on CERN groups and user e-mail."""
+
+    # Check if CERN user is logged in.
+    provides = session.get('identity.cern_provides')
```

Comment:
could we get this from the identify directly instead of loading it from the session? The identity should be already populated with ALL the user `provides` so you don't need to add the user email for example